Source/DafnyTestGeneration/TestMethod.cs

```diff
@@ -89,7 +103,11 @@ public class TestMethod {
       }
 
       List<string> elements = new();
-      switch (variable.Type.Name) {
+      var variableType = variable.Type.ReplaceTypeVariables(defaultType);
```

Comment:
This does not work for higher-kinded type variables, right? But I don't think we have them in Dafny.